docs/basics-in-depth/dependency_injection.md

```diff
@@ -0,0 +1,215 @@
+---
+id: dependency_injection
+title: Dependency Injection
+sidebar_label: Dependency Injection
+---
+
+# What is Dependency Injection?
+
+Dependency injection is a software design pattern in which a component's dependencies are provided by an external source rather than being hard-coded. This is a very powerful pattern that allows us to decouple our code and make it more testable. It also allows us to have a single source of truth for our dependencies.
+
+---
+
+# What is a service locator?
+
+The service locator a pattern in which an object looks up its dependency from a central `locator` object. This `locator` is responsible for creating and managing the lifecycle of the dependencies, and the object looks them up by the type or the name.
```

Comment:
- The service locator `*is*` a pattern
- You can stop the last sentence after dependencies and remove everything after the comma.